Add simple account deployment script

oneleo opened the pull request:
- Add simple account deployment script

oneleo committed:
doc: Modify environment variables in the deployment script.

oneleo committed:
doc: Add SimpleAccount deployment script

pilagod commented on README.md:
```diff
@@ -76,7 +76,7 @@ Transaction hash: 0x7a6d29b281aaef31e5d7ae5080b8800614f4f053b63fbb83a0d870807197
 PASSKEY_ACCOUNT_FACTORY_ADDRESS="<YOUR_PASSKEY_ACCOUNT_FACTORY_ADDRESS>"
 
-forge verify-contract --watch --chain "sepolia" --verifier "etherscan" --etherscan-api-key ${ETHERSCAN_API_KEY} --compiler-version ${COMPILER_VERSION} --constructor-args $(cast abi-encode "constructor(address)" ${ENTRYPOINT_ADDRESS}) ${PASSKEY_ACCOUNT_FACTORY_ADDRESS} "src/account/PasskeyAccountFactory.sol:PasskeyAccountFactory"
+forge verify-contract --watch --chain ${NETWORK_NAME} --verifier "etherscan" --etherscan-api-key ${ETHERSCAN_API_KEY} --compiler-version ${COMPILER_VERSION} --constructor-args $(cast abi-encode "constructor(address)" ${ENTRYPOINT_ADDRESS}) ${PASSKEY_ACCOUNT_FACTORY_ADDRESS} "src/account/PasskeyAccountFactory.sol:PasskeyAccountFactory"
```
According to the [doc](https://book.getfoundry.sh/reference/forge/forge-verify-contract#forge-verify-contract), chain can be assigned by id. In my opinion it could be better to use id instead of name.

pilagod commented on .env.deployment.example:
```diff
@@ -1,8 +1,11 @@
-NODE_RPC_URL=
+NETWORK_NAME=sepolia
+NODE_RPC_URL=https://ethereum-${NETWORK_NAME}.publicnode.com
 COMPILER_VERSION=0.8.23
```
```suggestion
COMPILER_VERSION=0.8.24
```

pilagod reviewed (commented):
In the future, it is clearer to organize these shell commands into files.

oneleo commented:
> In the future, it is clearer to organize these shell commands into files.

Perhaps we can consider how to improve, as verification may fail at times.

oneleo committed:
doc: Move the deployment script to deployment.md.

1. Adjust the environment variable name.
2. Adjust the deployment and verification scripts for future
   single-file execution.
